magda-web-client/src/helpers/ChartDatasetEncoder.js

```diff
@@ -12,47 +12,145 @@ import takeRight from "lodash/takeRight";
 import sortBy from "lodash/sortBy";
 import * as d3 from "d3-collection";
 import { config } from "../config";
+import chrono from "chrono-node";
 import type { ParsedDistribution } from "../helpers/record";
 
-let Papa = null;
-
-const avlChartTypes = ["bar", "pie", "scatter", "line"];
-const fetchData = function(url) {
-    return new Promise((resolve, reject) => {
-        Papa.parse(config.proxyUrl + "_0d/" + url, {
-            download: true,
-            header: true,
-            skipEmptyLines: true,
-            trimHeader: true,
-            complete: results => {
-                resolve(results);
-            },
-            error: err => {
-                let e;
-                if (!err)
-                    e = new Error("Failed to retrieve or parse the file.");
-                else e = err;
-                reject(e);
-            }
-        });
+const AVAILABLE_CHART_TYPES = ["bar", "pie", "scatter", "line"];
+const STRIP_NUMBER_REGEX = /[^\-\d.+]/g;
+
+/** Columns with these names are more likely to be picked as the default column on the x axis */
+const HIGH_PRIORITY_X_AXES = [
+    "gender",
+    "sex",
+    "occupation",
+    "state",
+    "city",
+    "company",
+    "postcode",
+    "category"
+];
+/** Columns with these names will not be picked as the default column on the x axis unless there's nothing else */
+const LOW_PRIORITY_X_AXES = ["name"];
+/** Columns with these names are more likely to be picked as the default column on the y axis */
+const HIGH_PRIORITY_Y_AXES = [];
+/** Columns with these names will not be picked as the default column on the y axis unless there's nothing else */
+const LOW_PRIORITY_Y_AXES = ["count", "id"];
+
+/** Columns with these terms in them will be overridden as category types */
+const OVERRIDE_TO_CATEGORY_COLUMNS = [
+    "id",
+    "code",
+    "identifier",
+    "postcode",
+    "sex",
+    "suburb",
+    "occupation",
+    "gender",
+    "abn",
+    "acn",
+    "afsl",
+    "pcode",
+    "month",
+    "reason",
+    "year"
+];
+/** Columns with these terms in them will be overridden as time types */
+const OVERRIDE_TO_TIME_COLUMNS = ["date", "time"];
+/** Columns with these terms in them will be overridden as numeric types */
+const OVERRIDE_TO_NUMERIC_COLUMNS = [
+    "amt",
+    "amount",
+    "sum",
+    "count",
```

Comment:
I think probably the regex expression (testKeywords) doesn't enforce the word edging (or string edge). So `country` is now set to a numeric column.  
One example: https://1140-preview-chart-dates.dev.magda.io/dataset/ds-dga-ab7eddce-84df-4098-bc8f-500d0d9776d1/details?q=*
A quick fix (if the regex is hard to fix) could be:
- remove  the `count` from `OVERRIDE_TO_NUMERIC_COLUMNS `
- or add `country` to `OVERRIDE_TO_CATEGORY_COLUMNS `

Reply:
I added the country to override category - I still want to have the regex super-greedy so it catches column names like `BLAHCOUNT` or `COUNTBLAH`

Reply:
@AlexGilleran 
Yeah, ideally the regex should be super-greedy to avoid miss matches. I had a few tries initially but give up due to the timelines --- one issue related to this is we probably want the regex greedy on three different boundaries:  non-word boundary, $ (ending string) & ^ (beginning of the string).
Second thought on this, I think it might be easier if we use regex to split the string to words and reformat them before the match a list of keyword.
Having said that, do you want to merge this PR for now (for the spirit of agile 😄 ) since all fixes are in place and continue the development on a new issue / new PR?